components/server/src/api/organization-service-api.ts

```diff
@@ -257,119 +254,64 @@ export class OrganizationServiceAPI implements ServiceImpl<typeof OrganizationSe
             throw new ApplicationError(ErrorCodes.BAD_REQUEST, "organizationId is required");
         }
 
-        const update: Partial<ProtocolOrganizationSettings> = {};
-        if (req.updateRestrictedEditorNames) {
-            update.restrictedEditorNames = req.restrictedEditorNames;
-        } else if (req.restrictedEditorNames.length > 0) {
+        const settings = req.settings;
```

Comment:
The code here was split into three things:
 1. gRPC validation (stayed here)
 2. general field validation (went into `organization-service.ts`
 3. field translation (went to `public-api-converter.ts`)